### ISSUE TITLE
Gonzales 3.3.5 issue with nth selector

### ISSUE THREAD

**Issue by travi** [sasstools/sass-lint#760]
What version of Sass Lint are you using?
1.8.1

Please include any relevant parts of your configuration

What did you do? Please include the actual source code causing the issue.
I use `grunt-sass-lint` which depends on a range (`^1.2.0`). Rather than specify a version in my project directly on `sass-lint`, I leverage it through the transitive dependency. When a build ran after the new version was released, the new version was pulled in, resulting in this new error.

What did you expect to happen?
I expected `^1.2.0` to be safe from breaking changes. With no failing lint errors before pulling in the new version, I expected no failing lint errors after pulling in the new version.

What actually happened? Please include any error messages given to you by Sass Lint.
I received a new fatal error that provides little detail about the actual problem.
```
157:1  error  Please check validity of the block starting from line #157  Fatal
```
the mentioned block is as follows:
```css
        &:nth-child(n+10) {
          visibility: hidden;
        }
```

If you're using a IDE plugin have you tried the CLI too?
N/A

**Comment by DanPurdy** [comment]
That error message means the breaking change isn't with sass-lint it's with our AST dependency [gonzales-pe](https://github.com/tonyganch/gonzales-pe)

**Comment by travi** [comment]
what options do i have as a consumer other than pinning an old version?

**Comment by DanPurdy** [comment]
Well gonzales also only released a patch version so i'll have a look into it, if nothing else we can roll a quick hotfix to pin gonzales at its previous version soon.

**Comment by travi** [comment]
thank you for digging in. i'll pin a version on my end for now. happy to provide more info if it would be helpful.

**Comment by DanPurdy** [comment]
Awesome, sorry for the breakages

**Comment by DanPurdy** [comment]
so it's the `n+10` causing the issue 

if you change it to this interestingly it works...
```
&:nth-child(1n+10) {
    visibility: hidden;
  }
```

or even 

```
&:nth-child(n+b) {
    visibility: hidden;
  }
```

I'll report it at gonzales and see if i can spot the issue, for now though I think i'll run a hotfix release 1.8.2 and pin gonzales back a version as this is obviously quite a common scenario.

**Comment by DanPurdy** [comment]
For reference https://github.com/tonyganch/gonzales-pe/issues/194

**Comment by DanPurdy** [comment]
waiting for the tests to pass in #761

**Comment by bgriffith** [comment]
Pushed a PR that fixes this issue :+1:

**Comment by DanPurdy** [comment]
This is fixed and parses correctly in sass-lint 1.9.0
